modules/irsa/external-deployments-operator.tf

```diff
@@ -1,23 +1,280 @@
+    resources = ["*"]
+  }
+  statement {
+    sid    = "LogsForSagemaker"
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogStreams",
+      "logs:GetLogEvents",
+      "logs:FilterLogEvents"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:logs:${local.region}:${local.account_id}:log-group:/aws/sagemaker/*"
+    ]
+  }
+  statement {
+    sid    = "SagemakerManageResources"
+    effect = "Allow"
+    actions = [
+      "sagemaker:AddTags",
+      "sagemaker:CreateEndpoint",
+      "sagemaker:CreateEndpointConfig",
+      "sagemaker:CreateModel",
+      "sagemaker:DeleteEndpoint",
+      "sagemaker:DeleteEndpointConfig",
+      "sagemaker:DeleteModel",
+      "sagemaker:DescribeEndpoint",
+      "sagemaker:DescribeEndpointConfig",
+      "sagemaker:DescribeModel",
+      "sagemaker:InvokeEndpoint",
+      "sagemaker:UpdateEndpointWeightsAndCapacities"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "AutoscalingForSagemaker"
+    effect = "Allow"
+    actions = [
+      "application-autoscaling:DeleteScalingPolicy",
+      "application-autoscaling:DeregisterScalableTarget",
+      "application-autoscaling:DescribeScalableTargets",
+      "application-autoscaling:DescribeScalingActivities",
+      "application-autoscaling:DescribeScalingPolicies",
+      "application-autoscaling:PutScalingPolicy",
+      "application-autoscaling:RegisterScalableTarget",
+      "application-autoscaling:TagResource"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "CloudwatchForAutoscaling"
+    effect = "Allow"
+    actions = [
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DeleteAlarms",
+      "cloudwatch:DescribeAlarms"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+  statement {
+    sid    = "IamAllowCreateServiceLinkedRole"
+    effect = "Allow"
+    actions = [
+      "iam:CreateServiceLinkedRole"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:iam::${local.account_id}:role/aws-service-role/sagemaker.application-autoscaling.amazonaws.com/*"
+    ]
+    condition {
+      test     = "StringLike"
+      variable = "iam:AWSServiceName"
+      values = [
+        "sagemaker.application-autoscaling.amazonaws.com"
+      ]
+    }
+  }
+  statement {
+    sid    = "EcrRegistryReadDominoEnvironments"
+    effect = "Allow"
+    actions = [
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:BatchGetImage",
+      "ecr:GetDownloadUrlForLayer",
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:ecr:${local.region}:${local.account_id}:repository/${local.environments_repository}",
+      "arn:${data.aws_partition.current.partition}:ecr:${local.region}:${local.account_id}:repository/${local.environments_repository}*"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "external_deployments_assume_any_role" {
```

Comment:
Why would this be used? Couldn't it just assume something with super broad access and do whatever?

Reply:
This is here so that the operator role can assume a different role to deploy endpoints in a different AWS account.  The other accounts role also has to grant access to this role to assume it.  This is just giving the domino was account permission to assume other roles if the corresponding policy is in place on the other side.
I figure there could be situations when this is too broad so that was why there's a flag to allow it to be disabled.
Lmk if you have any ideas about a better way to do this.  The main goal is that you can setup to deploy endpoints in a different AWS account without having to make changes to the domino aws account